packages/material-tree-renderer/example/index.ts

```diff
@@ -1,131 +1,120 @@
-import { LabelDefinition } from '../src/helpers/util';
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import { combineReducers, createStore, Store } from 'redux';
-import { detailSchemata, imageProvider, labelProvider, modelMapping } from './config';
-import { taskSchema } from './schema';
-import { materialFields, materialRenderers } from '@jsonforms/material-renderers';
 import * as _ from 'lodash';
-import { findAllContainerProperties, Property } from '../src/services/property.util';
-import { JsonSchema7 } from '@jsonforms/core';
-import { setContainerProperties, treeWithDetailReducer } from '../src/reducers';
 import * as JsonRefs from 'json-refs';
 import {
   Actions,
   jsonformsReducer,
-  RankedTester
+  JsonSchema7,
+  NOT_APPLICABLE,
+  traverse,
+  UISchemaElement,
+  UISchemaTester
 } from '@jsonforms/core';
+import { materialFields, materialRenderers } from '@jsonforms/material-renderers';
+import { taskSchema } from './schema';
+import { Property } from '../src/services/property.util';
 import App from './App';
+import { taskView, userGroupView, userView } from './uischemata';
 
 const uischema = {
   'type': 'MasterDetailLayout',
   'scope': '#'
 };
 
-const filterPredicate = (data: Object) => {
-  return (property: Property): boolean => {
-    if (!_.isEmpty(modelMapping) &&
-      !_.isEmpty(modelMapping.mapping)) {
-      if (data[modelMapping.attribute]) {
-        return property.schema.$id === modelMapping.mapping[data[modelMapping.attribute]];
-      }
-      return true;
-    }
-  };
-};
+const detailSchemata: { tester: UISchemaTester, uischema: UISchemaElement}[]  = [
+  {
+    tester: schema => _.keys(schema.properties).some(key => key === 'done') ?
+      1 : NOT_APPLICABLE,
+    uischema: taskView
+  },
+  {
+    tester: schema => _.keys(schema.properties).some(key => key === 'birthday') ?
+      1 : NOT_APPLICABLE,
+    uischema: userView
+  },
+  {
+    tester: schema => _.keys(schema.properties).some(key => key === 'users') ?
+      1 : NOT_APPLICABLE,
+    uischema: userGroupView
+  }
+];
+
+const filterPredicate = () => (property: Property): boolean =>
+  traverse(taskSchema, subSchema => subSchema === property.schema) !== undefined;
```

Comment:
Maybe instead of `===`, we can use `_.isEqual()` for deep comparison

Reply:
This always returned false, I'm wondering why this actually worked.

Reply:
Ah right, got it, `traverse` always returns a `boolean` (which is kinda weird, but that's a different story), hence the expression evaluated to true. I hopefully fixed this by tracking the schema path in `findContainerProps` as well and maintaining an additional property `orgSchema` in `Property` which is the reference to the sub schema of the root schema, so that we are able to compare by reference here.